$MirOS: ports/databases/sqlite/patches/patch-src_os_unix_c,v 1.1 2008/08/26 12:53:59 tg Exp $
--- src/os_unix.c.orig	Tue Aug  5 19:58:27 2008
+++ src/os_unix.c	Fri Mar 25 23:35:36 2016
@@ -2575,7 +2575,7 @@ static void *unixDlOpen(sqlite3_vfs *pVf
 ** error message.
 */
 static void unixDlError(sqlite3_vfs *pVfs, int nBuf, char *zBufOut){
-  char *zErr;
+  const char *zErr;
   enterMutex();
   zErr = dlerror();
   if( zErr ){
@@ -2663,6 +2663,17 @@ static int unixSleep(sqlite3_vfs *pVfs, 
 int sqlite3_current_time = 0;
 #endif
 
+#ifdef SQLITE_OMIT_DATETIME_FUNCS
+/* this must be reversible, as the OS' gmtime() and strftime() is used */
+#define uCTp(x) (x)
+#elif defined(__MirBSD__)
+/* convert time_t to POSIX' idea of it, for the internal algorithms */
+#define uCTp(x) timet2posix(x)
+#else
+/* assume POSIX */
+#define uCTp(x) (x)
+#endif
+
 /*
 ** Find the current time (in Universal Coordinated Time).  Write the
 ** current time and date as a Julian Day number into *prNow and
@@ -2672,11 +2683,11 @@ static int unixCurrentTime(sqlite3_vfs *
 #ifdef NO_GETTOD
   time_t t;
   time(&t);
-  *prNow = t/86400.0 + 2440587.5;
+  *prNow = uCTp(t)/86400.0 + 2440587.5;
 #else
   struct timeval sNow;
   gettimeofday(&sNow, 0);
-  *prNow = 2440587.5 + sNow.tv_sec/86400.0 + sNow.tv_usec/86400000000.0;
+  *prNow = 2440587.5 + uCTp(sNow.tv_sec)/86400.0 + sNow.tv_usec/86400000000.0;
 #endif
 #ifdef SQLITE_TEST
   if( sqlite3_current_time ){
